src/main/java/com/google/javascript/clutz/ImportRenameMapBuilder.java

```diff
@@ -1,36 +1,67 @@
 package com.google.javascript.clutz;
 
 import com.google.javascript.rhino.Node;
+import java.util.ArrayList;
 import java.util.Collection;
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 
 /**
  * If Clutz is running in incremental mode, closure can't resolve imported symbols, so it gives them
- * names in the current namespace (ie `module$contents$current$file`), instead of names that
- * reference the original namespace (ie `module$exports$imported$file`). ImportRenameMapBuilder
- * builds a map from the local name to the exported name so the original names can be substituted.
+ * names based on the symbol that the import was assigned to (either
+ * `module$contents$current$file_symbol` or just `symbol`) instead of names that reference the
+ * original namespace (ie `module$exports$imported$file`). ImportRenameMapBuilder builds a map from
+ * the local name to the exported name so the original names can be substituted.
  */
 public class ImportRenameMapBuilder {
 
   /**
-   * Build takes a collection of source files, parses them with the compiler, and walks the ast to
-   * find any `const variable = goog.require()` statements to build a map from the name closure
-   * generates in incremental mode to the exported name of the import.
+   * Build takes a collection of parsed inputs and walks the ast to find any imports into local
+   * variables to build a map from the name closure generates in incremental mode to the exported
+   * name of the import.
    */
   public static Map<String, String> build(Collection<Node> parsedInputs) {
     Map<String, String> importRenameMap = new HashMap<>();
     for (Node ast : parsedInputs) {
+      // Symbols can be imported into a variable in a goog.module() file, so look for imports in the
+      // body of the goog module.
       String moduleId = getGoogModuleId(ast);
-      // Closure can only put the symbol name into the current module's namespace, if the source
-      // file is a module, so if it isn't a module, just bail
       if (moduleId != null) {
-        importRenameMap.putAll(build(moduleId, ast));
+        importRenameMap.putAll(build(moduleId, ast.getFirstChild()));
+      }
+
+      // Or symbols can be imported into a variable in a goog.scope() block, so look for imports in
```

Comment:
top-level goog.scope block, techincally goog.scope can be used anywhere, but we dont want to count those.

Reply:
Ok.